File: .Net/RecognizersTextPerformanceTest/Services/LoggingService.cs
using RecognizersTextPerformanceTest.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RecognizersTextPerformanceTest.Services
{
    public class LoggingService
    {
        private List<ILogger> _loggers;

        public LoggingService()
        {
            _loggers = new List<ILogger>();
        }

        public void RegisterLogger(ILogger logger)
        {
            _loggers.Add(logger);
        }
        
        public void Log(string operationName, string message)
        {
            foreach (var logger in _loggers)
                logger.Log(operationName, message);
        }
    }
}

File: Python/services/logging_service.py
from models.Logger.console_logger import console_logger
from models.Logger.file_logger import file_logger

class logging_service:
    def __init__(self):
        self._loggers = []

    def add_logger(self, logger):
        self._loggers.append(logger)

    def log(self, operation_name, message):
        for logger in self._loggers:
            logger.log(operation_name, message)    